index.js

```diff
@@ -1,24 +1,46 @@
-let body = document.querySelector("#body")
-let h1 = document.createElement("h1")
-body.append(h1)
-h1.innerText = "Random Star Wars Ship Generator"
-let button = document.createElement("button")
-button.id = "button"
-body.append(button)
-button.innerText = "GET ME A SHIP!"
-let ul = document.createElement("ul")
-body.append(ul)
-let p = document.createElement("p")
-ul.append(p)
-
-button.addEventListener("click", function(evt) {
-    fetch("https:/swapi.dev/api/starships/")
-    .then (res => res.json())
-    .then (function(starship){
-        function randomNumber() {
-            let i = Math.floor(Math.random() * (10))
-            return i
-        }
-        p.innerText = starship.results[randomNumber()].name
-    })
-})
\ No newline at end of file
+function randomShipName(ships) {
+	const index = Math.floor(Math.random() * (ships.length)),
+		ship = ships[index];
+
+	if (ship)
+		return ship.name;
+	else
+		return 'Uh oh! Ship not found :[';
+}
+
+const body = document.querySelector('body'),
+	h1 = document.createElement('h1'),
+	button = document.createElement('button'),
+	p = document.createElement('p');
+
+let ships;
+
+// page title
+h1.innerText = 'Random Star Wars Ship Generator';
```

Comment:
I reversed the order of inserting the element, then giving it content. while both actions would happen in an instant, it's still a better practice to fully setup your elements before inserting them, since this will save on paints, which will actually be important at some point - I had to fix a production bug last week related to overly expensive paints crashing mobile browsers so it's real.